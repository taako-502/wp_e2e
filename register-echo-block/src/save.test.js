import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Save from './save'
import React from 'react'
import { create } from 'react-test-renderer'

// テストのセットアップ
global.React = React
Enzyme.configure({ adapter: new Adapter() })

test('こだまが表示される', () => {
  const wrapper = shallow(<Save attributes={{ text: 'こんにちは' }} />)
  const echos = wrapper.find('span')
  expect(echos).toHaveLength(4)
  echos.map(echo => expect(echo.text()).toBe('こんにちは'))
})

test('スナップショットが一致する', () => {
  const tree = create(<Save attributes={{ text: 'こんにちは' }} />).toJSON()
  expect(tree).toMatchSnapshot()
})
