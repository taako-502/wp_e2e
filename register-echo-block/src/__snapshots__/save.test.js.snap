// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショットが一致する 1`] = `
<div>
  <span
    className="value"
    style="font-size:1em"
  >
    こんにちは
  </span>
  <span
    style="font-size:.75em"
  >
    こんにちは
  </span>
  <span
    style="font-size:.5em"
  >
    こんにちは
  </span>
  <span
    style="font-size:.25em"
  >
    こんにちは
  </span>
  ...
</div>
`;
